### ISSUE TITLE
Sometimes invisible on launch in GNOME

### ISSUE THREAD

**Issue by personalizedrefrigerator** [MrCarroll/joplin-desktop-snap#24]
# Summary

At times, the Joplin window is invisible on launch in GNOME.

# Explanation

By default, Joplin [starts with a minimized window](https://github.com/laurent22/joplin/blob/75e6ae135b2f8b917426fe17e808419583acf5cf/packages/app-desktop/ElectronAppWrapper.ts#L106C4-L106C8). The window is unminimized after settings are loaded (provided that the "launch minimized" setting is disabled). On GNOME, this initial minimization causes https://github.com/laurent22/joplin/issues/8256.

Joplin thus disables this behavior when it detects that it's running on GNOME (using the `XDG_CURRENT_DESKTOP` environment variable [(code)](https://github.com/laurent22/joplin/blob/75e6ae135b2f8b917426fe17e808419583acf5cf/packages/lib/shim.ts#L64)).

In the snap application, `process.env.XDG_CURRENT_DESKTOP` returns `'Unity'` and thus Joplin starts minimized.

Would it be possible to set `XDG_CURRENT_DESKTOP` to a string containing `GNOME` when running on GNOME?

**Comment by JGCarroll** [comment]
I could override `XDG_CURRENT_DESKTOP` to Gnome unconditionally but I'd be concerned at the possibility this would cause problems later with e.g Wayland support where there's specific quirks between DE's that might cause complications. 

The snap shouldn't be changing the value though, if the value is showing as Unity it's because I expect the value outside the snap is legitimately Unity. Could you please try the following in a terminal for me?

```
james@james-Virtual-Machine:~$ echo $XDG_CURRENT_DESKTOP
GNOME
james@james-Virtual-Machine:~$ snap run --shell joplin-desktop
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

james@james-Virtual-Machine:/home/james$ echo $XDG_CURRENT_DESKTOP
GNOME
james@james-Virtual-Machine:/home/james$ 
exit
```

I'm hoping that both inside and outside the snap agree that Unity is the correct variable. The confusion then would be why Unity is showing rather than Gnome if you're not using Unity, but hopefully the above will help tell if the Snap is interfering with the variable.

**Comment by personalizedrefrigerator** [comment]
```
[primary@computer ~]$ echo $XDG_CURRENT_DESKTOP
ubuntu:GNOME
[primary@computer ~]$ snap run --shell joplin-desktop
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

primary@computer:/home/primary$ echo $XDG_CURRENT_DESKTOP
ubuntu:GNOME
primary@computer:/home/primary$ exit
exit
[primary@computer ~]$ 
```

In Joplin's development tools, however,
```
> process.env.XDG_CURRENT_DESKTOP
'Unity'
```

**Comment by personalizedrefrigerator** [comment]
Never mind! It looks like this is happening in the AppImage, too!

If I run  `export XDG_CURRENT_DESKTOP=GNOME` before starting `joplin-desktop`, `process.env.XDG_CURRENT_DESKTOP` is set to `GNOME` as expected, however.

I'm closing this issue ((it's upstream).

**Comment by JGCarroll** [comment]
For what it's worth, this seems to be a problem specifically with the Ubuntu modifications to Gnome, I.E, when the host `XDG_CURRENT_DESKTOP` is `ubuntu:GNOME`.

When I tested with Vanilla Gnome 3 without the Ubuntu modifications, the JS console returns "GNOME".

When I ran in the Ubuntu modified Gnome environment, I get "Unity" like yourself, even though the variable is `Ubuntu:GNOME`.

Whatever's mangling the variable is happening after the sandboxing so I'd agree that it's upstream. Hopefully this might help track what's going on.

Edit: Just see you've beaten me to it!
